ores/utilities/precached.py

```diff
@@ -66,142 +76,32 @@ def main(argv=None):
-            self.executor.submit(
-                self._score_revision, context_name, model_names, rev_id)
-
-        def _score_revision(self, context_name, model_names, rev_id):
-            if not model_names:
-                return
-            url = ores_url + "/v2/scores/" + context_name + "/" + \
-                  "?models=" + "|".join(model_names) + \
-                  "&revids=" + str(rev_id) + \
-                  "&precache=true"
-            try:
-                time.sleep(delay)
-                start = time.time()
-                response = requests.get(url, timeout=20, verify=True)
-                if response.status_code == 200:
-                    first_model_doc = \
-                        response.json()['scores'][context_name][model_names[0]]
-                    if 'error' not in first_model_doc['scores'][str(rev_id)]:
-                        metrics_collector.precache_score(
-                            context_name, model_names, time.time() - start)
-                    else:
-                        metrics_collector.precache_scoring_error(
-                            context_name, model_names, response.status_code,
-                            time.time() - start)
-                        logger.warning(
-                            "Request to {0} failed with {1} response: \n\t{2}"
-                            .format(url, response.status_code,
-                                    first_model_doc['scores'][str(rev_id)]))
-
-                else:
-                    metrics_collector.precache_scoring_error(
-                        context_name, model_names, response.status_code,
-                        time.time() - start)
-                    logger.warning(
-                        "Request to {0} failed with {1} response: \n\t{2}"
-                        .format(url, response.status_code, response.content[:500]))
-                if notify:
-                    watchdog_ping(*notify)
-            except requests.exceptions.ReadTimeout:
-                metrics_collector.precache_scoring_error(
-                    context_name, model_names, 504,
-                    time.time() - start)
-                logger.warning(
-                    "Request to {0} timed out in {1} seconds"
-                    .format(url, round(time.time() - start, 3)))
-            except Exception as e:
-                logger.error(str(type(e)) + ": " + str(e))
-
-    return _RCNamespace
+    # What to do in case of a change
+    def precache_a_change(change):
+        if delay:
+            time.sleep(delay)
+        if notify:
+            watchdog_ping(*notify)
+        start = time.time()
+        response = requests.post(
+            ores_url + "/v3/precache/", data=json.dumps(change))
+
+        if response.status == 200:
+            logger.info("Scored {0} in {1} seconds."
+                        .format(json.dumps(change), round(time.time() - start, 3)))
```

Comment:
I have a feeling that change will be a huge variable, we should only get rev id.